luigi/contrib/mrjob.py

```diff
@@ -0,0 +1,288 @@
+
+    def handle_counters(self, counters):
+        """ Override this if you want to do something with the counters
+        returned by `mrjob.runner.MRJobRunner.counters()`.
+        """
+        return
+
+    def __init__(self, *args, **kwargs):
+        super(MRJobTask, self).__init__(*args, **kwargs)
+        self.opts = self.make_mrjob_opts(*args, **kwargs)
+
+    def run(self):
+        # if the last step fails it's possible for mrjob to
+        # have created the target dir at least on s3
+        # without the actually being successfull. this causes an error and
+        # requires manual intervention. should we just delete the dir on s3 if
+        # it doesn't contain a _SUCCESS ?
+        job = self.job_cls(args=self.opts)
+        with job.make_runner() as jobrunner:
+            jobrunner.run()
+            # if user wanted output in a local file
+            if self.output_is_local_file:
+                with self.output().open('w') as output:
+                    output.writelines(jobrunner.stream_output())
+            self.handle_counters(jobrunner.counters())
+
+    def make_mrjob_opts(self, *args, **kwargs):
+        """ Makes a list of command line parameters to pass to the MRJob class
+        when instantiating. It makes sense to extend this behaviour e.g. if
+        you have some dynamic inputs that need to be passed as command line parameters.
+        """
+        base_opts = self.mrjob_opts()
+        base_opts.update((k, v) for k, v in kwargs.items() if k.startswith('-'))
+        opts_list = list(chain(*[_to_opts(k, v) for k, v in base_opts.items()]))
+        # now add special params: input and ouput
+        # yikes, existence of path field not really enforced
+        input_ = self.input()
+        if isinstance(input_, dict):
+            input_ = input_['input']
+        if not isinstance(input_, list):
+            input_ = [input_]
+        input_files = [
+                        _get_mrjob_friendly_path(in_)[0] for in_ in
+                        input_
+                       ]
+        output_dir, output_remote = _get_mrjob_friendly_path(self.output())
+
+        assert input_files and output_dir, \
+            'Must define output and dependencies or direct input'
+        # if it looks like output should go to a directory, add --no-output
+        # to avoid unnecessary streaming of potentially ginormous output
+        if (not output_remote and os.path.dirname(output_dir)+os.sep == output_dir)\
+                or output_remote:
+            self.output_is_local_file = False
+            opts_list += ['--no-output']
+            return opts_list + input_files + ['-o', output_dir]
+        elif not output_remote:
+            # if user wants the output to be a local file, enable streaming of output
+            # so we can write the file in runner
+            self.output_is_local_file = True
+            return opts_list + input_files
+
+
+class LocalMRJobTarget(luigi.file.LocalTarget):
```

Comment:
Can you inherit a more general target? What does LocalTarget give you over [FileSystemTarget](https://github.com/spotify/luigi/blob/831e390763d9b2c83faeef43985debf54d92403a/luigi/target.py#L203)?